### PR TITLE
CHANGES for edge-19.12.2

### PR DESCRIPTION
**Pull request by alpeb**
## edge-19.12.2

* CLI
  * Added support for injecting CronJobs and ReplicaSets, and the ability to use
    them as targets in the CLI subcomands
  * Added ability to update the trust anchor and issuer certificates during
    `linkerd upgrade` with the new flags `--identity-issuer-certificate-file`,
    `--identity-issuer-key-file` and `identity-trust-anchors-file`

* Controller
  * Fixed bug causing pods having security context capabilities to fail to be
    injected
  * Upgraded Prometheus to v1.2.1 (thanks @daxmc99!)

* Web UI
  * Added support for CronJobs and ReplicaSets, including new Grafana dashboards
    for them

* Proxy
  * Increased destination controller client's connection receive window to avoid
    it becoming exhausted, by single stalled streams
  * Eagerly read profiles off the wire

* Internal
  * Moved CNI template into a Helm chart, soon to be published
  * Reenabled certificates rotation integration tests

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+  * Upgraded Prometheus to v1.2.1 (thanks @daxmc99!)
+
+* Web UI
+  * Added support for CronJobs and ReplicaSets, including new Grafana dashboards
+    for them
+
+* Proxy
+  * Increased destination controller client's connection receive window to avoid
+    it becoming exhausted, by single stalled streams
+  * Eagerly read profiles off the wire
```
Comment (olix0r) [inline CHANGES.md]: These are sort of internals-focused -- the user facing impact here is that we believe we've addressed a long-standing service discovery staleness bug.

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,30 @@
+## edge-19.12.2
+
+* CLI
+  * Added support for injecting CronJobs and ReplicaSets, and the ability to use
+    them as targets in the CLI subcomands
+  * Added ability to update the trust anchor and issuer certificates during
+    `linkerd upgrade` with the new flags `--identity-issuer-certificate-file`,
+    `--identity-issuer-key-file` and `identity-trust-anchors-file`
+
```
Comment (scottcarol) [inline CHANGES.md]: Typically we don't have empty lines between sections

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,30 @@
+## edge-19.12.2
+
+* CLI
+  * Added support for injecting CronJobs and ReplicaSets, and the ability to use
+    them as targets in the CLI subcomands
```
Comment (scottcarol) [inline CHANGES.md]: subcommands

File: CHANGES.md [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+  * Added ability to update the trust anchor and issuer certificates during
+    `linkerd upgrade` with the new flags `--identity-issuer-certificate-file`,
+    `--identity-issuer-key-file` and `identity-trust-anchors-file`
+
+* Controller
+  * Fixed bug causing pods having security context capabilities to fail to be
+    injected
+  * Upgraded Prometheus to v1.2.1 (thanks @daxmc99!)
```
Comment (ihcsim) [inline CHANGES.md]: This is the Prometheus API (`client_golang`) package, not Prometheus itself. I think it can go under the _Internal_ section.

File: CHANGES.md [hunk 5]
```diff
@@ -1,3 +1,30 @@
...
+
+* CLI
+  * Added support for injecting CronJobs and ReplicaSets, and the ability to use
+    them as targets in the CLI subcomands
+  * Added ability to update the trust anchor and issuer certificates during
+    `linkerd upgrade` with the new flags `--identity-issuer-certificate-file`,
+    `--identity-issuer-key-file` and `identity-trust-anchors-file`
+
+* Controller
+  * Fixed bug causing pods having security context capabilities to fail to be
```
Comment (scottcarol) [inline CHANGES.md]: would it make more sense to change "having" to "with" ? TIOLI

File: CHANGES.md (hunk 1 above)
Comment (olix0r) [inline CHANGES.md]:
```suggestion
  *  Fixes a bug where the proxy could stop receiving service discovery updates, resulting in 503s.
```

File: CHANGES.md [hunk 6]
```diff
@@ -1,3 +1,30 @@
...
+  * Added support for CronJobs and ReplicaSets, including new Grafana dashboards
+    for them
+
+* Proxy
+  * Increased destination controller client's connection receive window to avoid
+    it becoming exhausted, by single stalled streams
+  * Eagerly read profiles off the wire
+
+* Internal
+  * Moved CNI template into a Helm chart, soon to be published
```
Comment (scottcarol) [inline CHANGES.md]: maybe instead of "soon to be published", "to prepare for future publication" so there's no time expectation?

File: CHANGES.md [hunk 7]
```diff
@@ -1,3 +1,30 @@
+## edge-19.12.2
+
+* CLI
+  * Added support for injecting CronJobs and ReplicaSets, and the ability to use
+    them as targets in the CLI subcomands
+  * Added ability to update the trust anchor and issuer certificates during
```
Comment (ihcsim) [inline CHANGES.md]: s/during/using

Or alternately,

> Introduce new flags to `linkerd upgrade` to support trust anchor and issuer certificate rotation

File: CHANGES.md [hunk 8]
```diff
@@ -1,3 +1,30 @@
+## edge-19.12.2
+
+* CLI
+  * Added support for injecting CronJobs and ReplicaSets, and the ability to use
```
Comment (kleimkuhler) [inline CHANGES.md]: Suggestion: ".. ConJobs and ReplicaSets, as well as the ability .."

File: CHANGES.md (hunk 5 above)
Comment (kleimkuhler) [reply CHANGES.md]: Maybe reword it: "Fixed inject failures for pods with security context capabilities"

File: CHANGES.md [hunk 9]
```diff
@@ -1,3 +1,30 @@
...
+    injected
+  * Upgraded Prometheus to v1.2.1 (thanks @daxmc99!)
+
+* Web UI
+  * Added support for CronJobs and ReplicaSets, including new Grafana dashboards
+    for them
+
+* Proxy
+  * Increased destination controller client's connection receive window to avoid
+    it becoming exhausted, by single stalled streams
```
Comment (kleimkuhler) [inline CHANGES.md]: Nit: Don't need the comma here.

File: CHANGES.md (hunk 4 above)
Comment (alpeb) [reply CHANGES.md]: Oops, good catch :+1:

File: CHANGES.md [hunk 10]
```diff
@@ -1,3 +1,24 @@
...
+  * Introduced the new flags `--identity-issuer-certificate-file`,
+    `--identity-issuer-key-file` and `identity-trust-anchors-file` to `linkerd
+    upgrade` to support trust anchor and issuer certificate rotation
+* Controller
+  * Fixed inject failures for pods with security context capabilities
+* Web UI
+  * Added support for CronJobs and ReplicaSets, including new Grafana dashboards
+    for them
+* Proxy
+  * Fixes a bug where the proxy could stop receiving service discovery updates,
```
Comment (olix0r) [inline CHANGES.md]: sorry sorry... tense check. this should be "fixed". my fault

Comment (olix0r) [reply CHANGES.md]:
```suggestion
  * Fixed a bug where the proxy could stop receiving service discovery updates,
```
